component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,378 @@
+                        attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+            this.groupKeyExpressionExecutor = attributeExpressionExecutors[1];
+            if (attributeExpressionExecutors.length == 3) {
+                this.groupKeyExpressionExecutor = attributeExpressionExecutors[1];
+                if ((attributeExpressionExecutors[2] instanceof ConstantExpressionExecutor)) {
+                    if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.INT) {
+                        timeInMilliSeconds = Integer.parseInt(String.valueOf(((ConstantExpressionExecutor)
+                                attributeExpressionExecutors[2]).getValue()));
+                    } else if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.LONG) {
+                        timeInMilliSeconds = Long.parseLong(String.valueOf(((ConstantExpressionExecutor)
+                                attributeExpressionExecutors[2]).getValue()));
+                    } else {
+                        throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter " +
+                                "'time.in.milliseconds' should be either be a constant (of type int or long), " +
+                                "but found " + attributeExpressionExecutors[2].getReturnType());
+                    }
+                } else {
+                    throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter " +
+                            "'time.in.milliseconds' should either be a constant (of type int or long)");
+                }
+            }
+        } else {
+            throw new SiddhiAppValidationException(
+                    "Resource batch window should only have two or three parameters, but found "
+                            + attributeExpressionExecutors.length + " input attributes");
+        }
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner) {
+        synchronized (this) {
+            long currentTime = siddhiAppContext.getTimestampGenerator().currentTime();
+            ComplexEventChunk<StreamEvent> outputStreamEventChunk = new ComplexEventChunk<StreamEvent>(true);
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+                StreamEvent clonedStreamEvent = streamEventCloner.copyStreamEvent(streamEvent);
+                if (!streamEvent.getType().equals(ComplexEvent.Type.TIMER)) {
+                    Object groupEventMapKey = groupKeyExpressionExecutor.execute(clonedStreamEvent);
+                    if (groupEventMap.get(groupEventMapKey) != null) {
+                        groupEventMap.get(groupEventMapKey).streamEventList.add(clonedStreamEvent);
+                    } else {
```

Comment:
Change it like
```
value = groupEventMap.get(groupEventMapKey);
if (value != null) {
       value.streamEventList.add(clonedStreamEvent);
}....
```

Reply:
Fixes added: 18ef02767fb180107b7938fefd40f4153d6f3807